Add GitHub Pullrequest documentation (#203)
updatecli/website: content/docs/plugins/github_pullrequest.adoc

---
title: "Github"
description: "Interact with Github PullRequest"
lead: "kind: github"
date: 2022-01-30T08:21:01+02:00
lastmod: 2022-01-30T08:21:01+02:00
draft: false
images: []
menu: 
  docs:
    parent: "plugins"
weight: 130 
toc: true
plugins:

>   - pullrequest

Author: @dduportal @lemeurherve For the time being, I am considering this "resource" a plugin for  "pull request". Once the "action" concept is ready to be used then we would just change this line to "action".
The only intends is to highlight this documentation in the correct doc menu

Reviewer: LGTM

Reviewer 2: Lgtm too , we are in sync

---

// <!-- Required for asciidoctor -->
:toc:
// Set toclevels to be at least your hugo [markup.tableOfContents.endLevel] config key
:toclevels: 4

== Description

> The pullrequest resource describes how updatecli handle GitHub pullrequest.

Reviewer 2: ```suggestion
The `pullrequests` section describes the link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests[GitHub Pull Requests] that updatecli is expected to open (or update) when the manifest is applied.
```

- It's not a resource ;)
- "describes how" is a wrong assertion: the whole goal of describing is to abstract the "how" away from user (vs. scripting)
- Not everyone is at ease with pull request: adding a link to official GitHub doc

== Parameters

[cols="1,1,1,4",options=header]
|===
| Name | Required | Default |Description

> | automerge | |-| Enable automerge. More on link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request[GitHub documentation]

Reviewer 2: ```suggestion
| automerge | |-| Enable the link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request[GitHub "auto-merge" feature] for the Pull Request
```

- Given that we discussed a lot about the expected meaning of wording "automerge", WDYT about making it explicit that it is the GitHub automerge feature?

Author: Thanks for the suggestio, I like it very much. Also thanks for the default value which I forgot to put

> | description | |-| Insert pullrequest text before the one generated by updatecli

Reviewer 2: Can you elaborate on this one, I don't understand what it does?

Reviewer 2: ```suggestion
| description | | "" | Insert pullrequest text before the one generated by updatecli
```

> | draft | |-| Disallow maintainer to modify the pullrequest

Reviewer 2: ```suggestion
| draft | |-| Set this pull request in a link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft[Draft state (as defined by GitHub)]
```

> | labels | |-| Set PullRequest labels

Reviewer 2: ```suggestion
| labels | |-| Set the labels of this pull request (list of labels)
```

> | automerge | | false | Enable the link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request[GitHub "auto-merge" feature] for the Pull Request

Reviewer: ```suggestion
| automerge | | false | Enable the link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request[GitHub "auto-merge" feature] for the pull request
```

> | maintainercannotmodify | |-| Disallow maintainer to modify the pullrequest

Reviewer 2: Did not know about this one. Is there more details on GitHub documentation to help understanding?

Reviewer: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Reviewer 2: oh thanks! then:

```suggestion
| maintainercannotmodify | |-| Forbid maintainers to link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests[edit the pull request]
```

> | title | |-| Set pullrequest title

Reviewer 2: ```suggestion
| title | |-| Set the title of the pull request
```

Reviewer: I prefer the shorter version, is there a requirement for the longer one?

Reviewer 2: > I prefer the shorter version, is there a requirement for the longer one?

none except "pull request in 2 words

Reviewer 2: ```suggestion
| title | |-| Set the pull request title
```

WDYT?

Reviewer: I think "pull request" in 2 words should be applied everywhere then :)

Reviewer 2: Yes, i agree

Reviewer 2: (Except on the yaml of course)

> | description | | "" | Insert pullrequest text before the one generated by updatecli

Reviewer: ```suggestion
| description | | "" | Insert the pull request text before the one generated by updatecli
```

Reviewer: I find this not really clear.

Author: What would you think of:
```suggestion
| description | | "" | Prepend a custom message to the pull request body generated by updatecli
```

Reviewer: ```suggestion
| description | | "" | Prepend a custom description to the pull request body message generated by updatecli
```

Reviewer 2: 👍  (both proposals)

Author: I prefer @lemeurherve 's one

> | draft | | false | Set this pull request in a link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft[Draft state (as defined by GitHub)]

Reviewer: ```suggestion
| draft | | false | Set this pull request in a link:https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft[draft state (as defined by GitHub)]
```
